### DISCUSSION TITLE
Regarding Qdrant's data storage disk-related issues.

### DISCUSSION THREAD

**Discussion by wanghaiyongyw** [qdrant/qdrant#3734] [General]
I would like to inquire whether, when Qdrant updates a vector (using the same primary key ID), repeatedly writing data will occupy multiple data (disk) spaces. This is because it has been observed that when updating the vector multiple times using the same primary key ID, the disk space increases significantly.

**Comment by generall** [comment]
Usually what happens is old vector is marked as removed and new vector created instead. That happens until the specified threshold is reached and the optimizer clears all the outdated records.

**Reply by yangchen625** [reply]
When does the optimizer start clearing stale records?
